LabExT/Movement/PathPlanning.py

```diff
@@ -284,3 +286,120 @@ def _find_lowest_potential(self) -> np.ndarray:
                 curr_idx = possible_idx
 
         return curr_idx
+
+
+class LocalMinimumError(RuntimeError):
+    pass
+
+
+class PathPlanning:
+    """
+    Main class for Path Planning
+    """
+
+    def __init__(self, chip) -> None:
+        """
+        Constructor for the Path Planning.
+
+        Parameters
+        ----------
+        chip: Chip
+            Instance of the a chip
+        """
+        self.chip: Type[Chip] = chip
+        self.grid_size, self.grid_outline = self._get_grid_properties(tol=100)
+
+        self.potential_fields = {}
+        self.last_moves = []
+
+    def set_stage_target(
+            self,
+            calibration: Type["Calibration"],
+            target: Type[ChipCoordinate]) -> None:
+        """
+        Registers a target for the given calibration.
+
+        Parameters
+        ----------
+        calibration : Calibration
+            Instance of a calibration
+        target : ChipCoordinate
+            Target of the stage in chip coordinates.
+        """
+        self.potential_fields[calibration] = PotentialField(
+            calibration,
+            target,
+            self.grid_size,
+            self.grid_outline)
+
+    def trajectory(self, abort_local_minimum=3):
+        """
+
+        """
+        while not all(f.current_position ==
+                      f.target_position for f in self.potential_fields.values()):
+
+            next_move = {}
+
+            for calibration, potential_field in self.potential_fields.items():
+                potential_field.set_stage_obstacles(*[
+                    obstacle_calibration for obstacle_calibration in self.potential_fields.keys() if obstacle_calibration != calibration
+                ])
+
+                next_move[calibration] = potential_field.next_waypoint()
+
+            if self._last_moves_equal(
+                    self.last_moves[-abort_local_minimum:], next_move):
+                raise LocalMinimumError
+
+            self.last_moves.append(next_move)
+            yield next_move
+
+    def _get_grid_properties(self, tol: float = 0) -> tuple:
```

Comment:
call the `tol` parameter `padding`, as its not a tolerance but a "pad" around the maximum chip coordinates found.

Reply:
Good idea!